### PR TITLE
#23253 Lint for valid CSS flags

### PR DESCRIPTION
**Pull request by DanielRyanSmith**
#23253
Running the lint will now check the CSS flags in the source file and check if they are valid or deprecated. If any are found that are invalid, the lint will now show an error and ask for a fix. This check is only done for test in the `css/` directory.

CSS requirement flags pulled from the docs [here](http://web-platform-tests.org/writing-tests/css-metadata.html#requirement-flags).

### CODE REVIEW

**Comment by DanielRyanSmith** [comment]
It looks like the documentation build failing might be related to [this issue here](https://github.com/sphinx-doc/sphinx/issues/9841) since the same warning is showing and are using docutils-0.18.

File: tools/lint/constants.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        'HTMLonly',
+        'invalid',
+        'may',
+        'nonHTML',
+        'paged',
+        'scroll',
+        'should',
+        'svg'
+    }),
+    'deprecated': frozenset({
```
Comment (foolip) [inline tools/lint/constants.py:14]: How many do we have of these? Can they be removed instead of adding them to the lint?

Comment (DanielRyanSmith) [reply tools/lint/constants.py:14]: Based on [this run](https://community-tc.services.mozilla.com/tasks/IDor1Du-RimCuMg0BDuAog/runs/0/logs/public/logs/live.log) we have:
`INFO:lint:There were 6907 errors (DEPRECATED-CSS-REQUIREMENT-FLAG: 1066 INVALID-CSS-REQUIREMENT-FLAG: 5841)`
We can begin removing the invalid or deprecated flags as well. If we remove the existing flags, would it not be worth keeping a check in the lint to make sure no more invalid flags are added?

**Comment by DanielRyanSmith** [comment]
This change will require a larger cleanup of the flags currently in the repository to be landed, and with that comes more work that will require further planning. This is a low-priority change with a high amount of work involved. Closing this for now.

**Pull request closed without merging**